sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/CreateAnnouncementFragment.java

```diff
@@ -66,17 +105,76 @@ public void onStop() {
+                if (mAnnouncementCategory.getSelectedItem().equals("General Announcement")) {
+                    mAnnouncement.setCategory(1);
+                    mAnnouncement.setUserId(mUser.getId());
+                    mAnnouncement.setUserName(mUser.getName());
+                    mAnnouncement.setSchoolId(0);
+                    mAnnouncement.setSchoolName(null);
+                    mAnnouncementSchoolList.setVisibility(View.GONE);
+                } else if (mAnnouncementCategory.getSelectedItem().equals("School Announcement")) {
+                    mAnnouncement.setCategory(0);
+                    mAnnouncementSchoolList.setVisibility(View.VISIBLE);
+                    mSchoolAdapter = new SchoolSpinnerAdapter(getActivity(), mSchools, R.layout.simple_spinner_item, R.layout.simple_spinner_item);
+                    mSchoolAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+                    mAnnouncementSchoolList.setAdapter(mSchoolAdapter);
+                    mAnnouncementSchoolList.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+                        @Override
+                        public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+                            School school = (School) adapterView.getSelectedItem();
+                            mAnnouncement.setSchoolId(school.getId());
+                            mAnnouncement.setSchoolName(school.getName());
+                            mAnnouncement.setUserId(mUser.getId());
+                            mAnnouncement.setUserName(mUser.getName());
+                        }
+
+                        @Override
+                        public void onNothingSelected(AdapterView<?> adapterView) {
+//                            School school = (School) adapterView.getSelectedItem();
+//                            mAnnouncement.setSchoolId(school.getId());
+//                            BaseInterface baseInterface = (BaseInterface) getActivity();
+//                            mUser = baseInterface.getUser();
+//                            mAnnouncement.setUserId(mUser.getId());
+                        }
+                    });
+                }
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {
+//                mAnnouncement.setCategory(5);
+//                mAnnouncement.setUserId(mUser.getId());
+            }
+        });
     }
 
-    private void submitRequest() {
+    private boolean isValid() {
+        return mValidator.hasNonBlankField(mAnnouncementTitle, "Title") &
+                mValidator.hasNonBlankField(mAnnouncementBody, "Body");
+    }
 
-//        Requests.Announcements.with(getActivity()).makeCreateRequest(   );
+    private void validateAndSubmitRequest() {
+        if (!isValid())
+            return;
+        mAnnouncement.setTitle(mAnnouncementTitle.getText().toString());
```

Comment:
you want to do all your setting here, instead of in your selected item listener. you can get spinner.getItemPosition() && spinner.getSelectedItem() to get the category && school_id, respectively